src/Compilers/Core/Portable/Operations/OperationNodes.cs

```diff
@@ -8077,6 +8196,230 @@ public override IOperation Guard
         }
     }
 
+    /// <summary>
+    /// Represents a C# switch expression.
+    /// </summary>
+    internal abstract partial class BaseSwitchExpressionOperation : Operation, ISwitchExpressionOperation
+    {
+        public BaseSwitchExpressionOperation(
+            ITypeSymbol type,
+            SemanticModel semanticModel,
+            SyntaxNode syntax,
+            bool isImplicit)
+            : base(OperationKind.SwitchExpression, semanticModel, syntax, type: type, constantValue: default, isImplicit)
+        {
+        }
+        public abstract IOperation Value { get; }
+        public abstract ImmutableArray<ISwitchExpressionArmOperation> Arms { get; }
+        public override IEnumerable<IOperation> Children
+        {
+            get
+            {
+                yield return Value;
+                foreach (var arm in Arms)
+                {
+                    yield return arm;
+                }
+            }
+        }
+        public override void Accept(OperationVisitor visitor)
+        {
+            visitor.VisitSwitchExpression(this);
+        }
+        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
+        {
+            return visitor.VisitSwitchExpression(this, argument);
+        }
+    }
+
+    internal sealed partial class SwitchExpressionOperation : BaseSwitchExpressionOperation
+    {
+        public SwitchExpressionOperation(
+            ITypeSymbol type,
+            IOperation value,
+            ImmutableArray<ISwitchExpressionArmOperation> arms,
+            SemanticModel semanticModel,
+            SyntaxNode syntax,
+            bool isImplicit)
+            : base(type, semanticModel, syntax, isImplicit)
+        {
+            this.Value = value;
```

Comment:
>this.Value = value; [](start = 12, length = 19)

The established pattern is that parent is assigned before or as part of initializing a property. For example:
```
this.Value = SetParentOperation(value, this);
```

Consider following the pattern.